api.go

```diff
@@ -2892,9 +2865,39 @@ func (sc *Client) GetSitePrimaryAgentKey(corpName, siteName string) (PrimaryAgen
 	return primaryKey, nil
 }
 
+func (sc *Client) doRequestDetailed(method, url, reqBody string) (*http.Response, error) {
+	client := &http.Client{}
+
+	var b io.Reader
+	if reqBody != "" {
+		b = strings.NewReader(reqBody)
+	}
+
+	req, _ := http.NewRequest(method, apiURL+url, b)
+
+	if sc.email != "" {
+		// token auth
+		req.Header.Set("X-API-User", sc.email)
+		req.Header.Set("X-API-Token", sc.token)
+	} else {
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", sc.token))
+	}
+
+	if sc.fastlyKey != "" {
+		req.Header.Set("Fastly-Key", sc.fastlyKey)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Set("User-Agent", "go-sigsci")
+
+	resp, err := client.Do(req)
+
+	return resp, err
+}
+
 // CreateOrUpdateEdgeDeployment initializes the Next-Gen WAF deployment in Compute@Edge and configures the site for Edge Deployment.
 func (sc *Client) CreateOrUpdateEdgeDeployment(corpName, siteName string) error {
-	_, err := sc.doRequest("PUT", fmt.Sprintf("/v0/corps/%s/sites/%s/edgeDeployment", corpName, siteName), "")
+	_, err := sc.doRequestDetailed("PUT", fmt.Sprintf("/v0/corps/%s/sites/%s/edgeDeployment", corpName, siteName), "")
```

Comment:
Shouldn't this still be calling `doRequest()`? If it needs `doRequestDetailed`, maybe include a comment for why.

Reply:
Resolved by @vvuksan in https://github.com/signalsciences/go-sigsci/pull/55/commits/e93edecbc57a61eeb548c380aea710f38fd7a9db